### PR TITLE
feat(safety_check): add hysteresis factor in safety check logic

### PR DESCRIPTION
**Pull request by satoshi-ota**
### <samp>🤖 Generated by Copilot at de27076</samp>

Implemented a new hysteresis mechanism for the avoidance module in the behavior path planner. This mechanism uses a `hysteresis_factor_safe_count` and a `hysteresis_factor_expand_rate` to avoid switching paths too frequently and to adjust the safety margin for the collision check. Renamed and added parameters in `avoidance.param.yaml` and related files to reflect this change.

## Tests performed

- [ ] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/747f916d-61bf-5818-9a39-874bda4af4e4?project_id=prd_jt)

## Effects on system behavior

improve avoidance safety check judgement

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/utils/path_safety_checker/safety_check.cpp [hunk 1]
```diff
@@ -273,8 +273,8 @@ bool checkCollision(
     const auto min_lon_length =
       calcMinimumLongitudinalLength(front_object_velocity, rear_object_velocity, rss_parameters);
 
-    const auto & lon_offset = std::max(rss_dist, min_lon_length);
-    const auto & lat_margin = rss_parameters.lateral_distance_max_threshold;
+    const auto & lon_offset = std::max(rss_dist, min_lon_length) * hysteresis_factor;
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/src/utils/path_safety_checker/safety_check.cpp:276]: @satoshi-ota 
Is it enough to consider same factor value for longtudinal/lateral direction?
I mean I don't have any specific concern about this, but if you have knowledge in this point, i just want to know!

**Review by kyoichi-sugahara — APPROVED** [review]
Super thanks so much for quick implemtentation!!!!!
LGTM

File: planning/behavior_path_planner/src/utils/path_safety_checker/safety_check.cpp (hunk 1 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/utils/path_safety_checker/safety_check.cpp:276]: @kyoichi-sugahara Thanks for your comment. I think there is a possibility that we need to separate lat/lon hysteresis factor in near future. But for now, I could improve safety check chattering without that. So, I'm going to add new param if it needs separated hysteresis factor.

Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/src/utils/path_safety_checker/safety_check.cpp:289]: @satoshi-ota 
I totally agree with your idea!
Thanks for the reply.

**Pull request merged**